appmap/_implementation/configuration.py

```diff
@@ -6,19 +6,59 @@
 import os
 import sys
 from contextlib import contextmanager
-from functools import wraps
+from functools import wraps, cached_property
 import time
 
 import yaml
 
 from . import env
 from . import event
 from .recording import Recorder, Filter
-from .utils import appmap_tls, split_function_name
+from .utils import appmap_tls, split_function_name, fqname
 
 logger = logging.getLogger(__name__)
 
 
+class Config:
+    """ Singleton Config class """
+    _instance = None
+
+    def __new__(cls):
+        if cls._instance is None:
+            logger.debug('Creating the Config object')
+            cls._instance = super(Config, cls).__new__(cls)
+
+            # If we're not enabled, no more initialization needs to be
+            # done.
+            cls._instance._initialized = not env.enabled()
+
+        return cls._instance
+
+    def __init__(self):
+        if self.__getattribute__('_initialized'):  # keep pylint happy
+            return
+
+        config_file = os.getenv("APPMAP_CONFIG", "appmap.yml")
+        with open(config_file) as file:
```

Comment:
@apotterri is error handling required if appmap.yml does not exist?

Reply:
We only try to load appmap.yml if `APPMAP=true` is in the environment. If it doesn't exist, you get a stack trace that ends with 
```
FileNotFoundError: [Errno 2] No such file or directory: 'appmap.yml'
```
which seems fine to me (and matches the behavior of appmap-ruby).